### ISSUE TITLE
Missing support for async functions? (Node 8)

### ISSUE THREAD

**Issue by Fr8Traindb** [testdouble/testdouble.js#284]
td.replace and td.object do not seem to handle Node 8 async functions.

Example:

```
var fish = {
    eat: async function () { },
    swim: function () { },
    details: {
      age: 10,
      name: 'goldie'
    }
  };
  var fish = td.object(fish);
```

results of fish:
        { eat: {},
          swim: { [Function: testDouble] toString: [Function] },
          details: { age: 10, name: 'goldie' },
          toString: [Function] }

The async function eat is not a testDouble function - it gets replaced with a empty object.  I get the same results using td.replace().

**Comment by searls** [comment]
We haven't looked at this yet, and they are not yet supported

**Comment by rosston** [comment]
@Fr8Traindb If you run `npm ls lodash` in your project, does it show something <4.17.0 under `testdouble`? I've tested `td.object` running on top of lodash >= 4.17.4, and async functions get detected properly.

**Comment by Fr8Traindb** [comment]
@rosston I can hardly believe that did it!  Upgraded to lodash 4.17.4 and all is good indeed!  I was on 4.16.2

**Comment by searls** [comment]
landed in 3.2.5
